src/sweeper/__main__.py

```diff
@@ -4,60 +4,105 @@
-        backup.backup_data(os.path.join(args['--workspace'], args['<table_name>']), args['--backup-to'])
+        backup.backup_data(os.path.join(args['--workspace'], args['--table-name']), args['--backup-to'])
 
+    #: check what quality check to run.
     if args['duplicates']:
-        #: do something
-        print(args)
-        # duplicates.find(args['--input'])
-        sweeper = DuplicateTest(args['--workspace'], args['<table_name>'])
+        call_sweeper(args, "DuplicateTest")
     elif args['invalids']:
-        #: do something
-        print(args)
+        call_sweeper(args, "InvalidsTest")
     elif args['empties']:
-        print(args)
+        call_sweeper(args, "EmptyTest")
+    elif args['all-checks']:
+        call_sweeper(args, "DuplicateTest")
+        call_sweeper(args, "EmptyTest")
+
+
+def call_sweeper(args, sweeper_test):
+    '''
+    orchestrate the sweeper calls.
 
-        sweeper = EmptyTest(args['--workspace'], args['<table_name>'])
+    args: the docopt arguments.
+    sweeper_test: the name of the sweeper function to call.
+    '''
+    print(args)
+
+    #: if table name was provided, then run quality check on single table.
+    if args['--table-name']:
+        #: call the quality check.
+        if sweeper_test == "DuplicateTest":
+            sweeper = DuplicateTest(args['--workspace'], args['--table-name'])
+            save_report(args, sweeper, args['--table-name'], 'duplicates')
+        elif sweeper_test == "EmptyTest":
+            sweeper = EmptyTest(args['--workspace'], args['--table-name']) 
+            save_report(args, sweeper, args['--table-name'], 'empties')
+        #: attempt to fix flagged items.
+        if args['--try-fix']:
+            sweeper.try_fix()
+    #: if table name was not provided, then run quality check on all tables in the workpace.
+    else:
+        #: get a list of feature classes that are contained in the workspace.
+        featureclasses = workspace_info.get_featureclasses(args['--workspace'])
+        for fc in featureclasses:
+            print(fc)
+            #: call the quality check.
+            if sweeper_test == "DuplicateTest":
+                sweeper = DuplicateTest(args['--workspace'], fc)
+                save_report(args, sweeper, fc, 'duplicates')
+            elif sweeper_test == "EmptyTest":
+                sweeper = EmptyTest(args['--workspace'], fc)
+                save_report(args, sweeper, fc, 'empties')
+            #: attempt to fix flagged items.
+            if args['--try-fix']:
+                sweeper.try_fix()
+
+
+def save_report(args, sweeper, table_name, quality_check):
+    '''
+    save the quality check report or print it to console.
+
+    args: the docopt arguments.
+    sweeper: the sweeper variable returned from the class.
+    table_name: the name of the feature class table.
+    quality_check: the name of the quality check (example: 'duplicates', 'empties').
+    '''
 
     report_data = sweeper.sweep()
```

Comment:
are we really calling sweeper.sweep in the save report method?

Reply:
i'm not sure on this one... i'm gonna have to punt to @eneemann  or @ZachBeck   do either of you know if it needs to be called that way?

Reply:
No.  We are calling it in main.py.  The .sweep method returns self.report, which is the dictionary that the report is built from.  This is saved as a variable and then passed into the .save_report method.

Reply:
This code has it in the save report method so maybe some copy pasting got fouled up?

Reply:
Oh, I guess Greg had to move it inside that function with the reorganization.  Because the save_report method now has args/sweeper passed into it.  Then the report_data variable is populated when the sweeper.sweep() line is executed.  I guess at that point we could just call it via "sweeper.report"?

Reply:
greg and I worked through a big refactor. This will be changing when he pushes